File: packages/ipc/src/Keychain.ts
import type { Keychain as IKeychain } from "@cortex/platform"
import { invoke } from "@tauri-apps/api/core"

export class Keychain implements IKeychain {
	async get(key: string): Promise<string | null> {
		return await invoke<string | null>("keychain_get", { key })
	}

	async set(key: string, value: string): Promise<void> {
		await invoke<void>("keychain_set", { key, value })
	}

	async delete(key: string): Promise<void> {
		await invoke<void>("keychain_delete", { key })
	}
}

File: apps/desktop/src-tauri/src/commands/keychain.rs
use crate::keychain;

#[tauri::command]
pub fn keychain_set(key: String, value: String) -> Result<(), String> {
    keychain::set(&key, &value)
}

#[tauri::command]
pub fn keychain_get(key: String) -> Result<Option<String>, String> {
    keychain::get(&key)
}

#[tauri::command]
pub fn keychain_delete(key: String) -> Result<(), String> {
    keychain::delete(&key)
}
